### PR TITLE
Merge Path Generation Code

### PR DESCRIPTION
**Pull request by Lytigas**
The stuff that needs to be there is there to begin trajectory generation.

### CODE REVIEW

**Review by inquisitivecrystal — APPROVED** [review]
I've been working on this for far too long, so I'm tired and I'll be brief. I've cleaned up most of the code style problems I saw. I need to do another pass, but the main problems I saw were that there were things that need documentation (one class, some methods, and some parameters I didn't know how to fill in). The test classes could also use some work. @Danappelxx should go over them, but they seem to me not to be very idiomatic. There are places where things should be separated out into different test cases, and other places where exceptions should just be allowed to reach the test runner, which knows how to handle them.

**Review by inquisitivecrystal — CHANGES_REQUESTED** [review]
Oops, meant request changes.

File: src/main/java/org/team114/lib/pathgenerator/ParametricOneVariableSpline.java [hunk 1]
```diff
@@ -6,12 +6,12 @@
      * @param t The parameter from 0 to 1 inclusive, representing the percent through the spline.
      * @return The value of the spline.
      */
-    public double getValueAt(double t);
+    double at(double t);
```
Comment (Lytigas) [inline src/main/java/org/team114/lib/pathgenerator/ParametricOneVariableSpline.java]: I don't know if conventions have evolved yet, but interfaces allow private methods in Java 9, so the explicit/implicit thing applies.

Comment (jmeggitt) [reply src/main/java/org/team114/lib/pathgenerator/ParametricOneVariableSpline.java]: The public is optional but I personally prefer it for readability and I don't think changing getValueAt to at is very beneficial. Changing it to eval though to match Polynomial may be an option.

Comment (bookofportals) [reply src/main/java/org/team114/lib/pathgenerator/ParametricOneVariableSpline.java]: Public, okay. Intelij doesn't like it, but that's survivable. GetValueAt is too verbose. At means literally the same thing. If eval is really more appropriate, it's reasonable too.

Comment (bookofportals) [reply src/main/java/org/team114/lib/pathgenerator/ParametricOneVariableSpline.java]: Oh, and could we rename this interface to something shorter?

Comment (jmeggitt) [reply src/main/java/org/team114/lib/pathgenerator/ParametricOneVariableSpline.java]: I like the name as is, and I can't really think of a name that doesn't really butcher the meaning/readability. The only thing I can think of is ParametricOneDSpline.

**Comment by bookofportals** [comment]
Also, I feel uncomfortable squashing this. It has to much development history, with substantial contributions from two developers. We could perhaps squash together some related commits, if the current history seems too messy.

File: src/main/java/org/team114/lib/pathgenerator/ParametricOneVariableSpline.java (hunk 1 above)
Comment (bookofportals) [reply src/main/java/org/team114/lib/pathgenerator/ParametricOneVariableSpline.java]: ParametricMonoSpline?

Comment (jmeggitt) [reply src/main/java/org/team114/lib/pathgenerator/ParametricOneVariableSpline.java]: Kind of makes it sound like a new type of spline.

Comment (bookofportals) [reply src/main/java/org/team114/lib/pathgenerator/ParametricOneVariableSpline.java]: Is there anything else specific it could reasonably mean?

**Review by Danappelxx — APPROVED** [review]
Still needs cleanup but not enough to be a show-stopper IMO. LGTM as far as I'm concerned.

**Comment by jmeggitt** [comment]
I think cleanup is a second priority. We just need to be able to use it in ocelot at the moment.

**Review by jmeggitt — APPROVED** [review]

**Pull request merged**